packages/mui-base/src/Dialog/Root/useDialogRoot.ts

```diff
@@ -0,0 +1,119 @@
+import * as React from 'react';
+import type { UseDialogRootParameters, UseDialogRootReturnValue } from './DialogRoot.types';
+import { useControlled } from '../../utils/useControlled';
+
+/**
+ *
+ * Demos:
+ *
+ * - [Dialog](https://mui.com/base-ui/react-dialog/#hooks)
+ *
+ * API:
+ *
+ * - [useDialogRoot API](https://mui.com/base-ui/react-dialog/hooks-api/#use-dialog-root)
+ */
+export function useDialogRoot(parameters: UseDialogRootParameters): UseDialogRootReturnValue {
+  const {
+    open: openParam,
+    defaultOpen = false,
+    onOpenChange,
+    type = 'dialog',
+    modal = true,
+    softClose = false,
+    onNestedDialogOpen,
+    onNestedDialogClose,
+  } = parameters;
+
+  const [open, setOpen] = useControlled({
+    controlled: openParam,
+    default: defaultOpen,
+    name: 'DialogRoot',
+  });
+
+  const [titleElementId, setTitleElementId] = React.useState<string | undefined>(undefined);
+  const [descriptionElementId, setDescriptionElementId] = React.useState<string | undefined>(
+    undefined,
+  );
+  const [popupElementId, setPopupElementId] = React.useState<string | undefined>(undefined);
```

Comment:
We could use a single `useId()` call and then do `${id}-title`? In `Popover` I considered just using `hasTitle`, `hasDescription`, given they're optional, and then use the derived id to set the aria attributes.

Reply:
Developers can always provide their own ids to subcomponents, so we need to register them in the root.